File: misc/python/chefAndAdventures.py
for _ in range(int(input())):
  n, m, x, y = map(int, input().split())
  n -= 1
  m -= 1
  if(n < 2 and n == m):
    print("Chefirnemo")
  elif(n % x < 2 and n % x == m % y):
    print("Chefirnemo")
  else:
    print("Pofik")

File: misc/cpp/ChefAndAdventures.cpp
#include<iostream>
using namespace std;

int main() {
	int t;
	int n, m, x, y;
	cin >> t;
	while (t--) {
		cin >> n >> m >> x >> y;
		if ((n - 1) % x == 0 && (m - 1) % y == 0) {
			cout << "Chefirnemo";
		} else if ((n - 2) % x == 0 && (m - 2) % y == 0) {
			cout << "Chefirnemo";
		} else {
			cout << "Pofik";
		}
		cout << endl;
	}
	return 0;
}